### PR TITLE
drivers: i2c: i2c_dw: Added intel lpss dma support for I2C

### PR DESCRIPTION
**Pull request by bindushree98**
Enabled intel LPSS DMA interface using dw common to support usage of internal DMA in LPSS I2C to transfer and
receive data.

### CODE REVIEW

File: dts/bindings/i2c/i2c-controller.yaml [hunk 1]
```diff
@@ -9,10 +9,8 @@ bus: i2c
 
 properties:
   "#address-cells":
-    required: true
```
Comment (tbursztyka) [inline dts/bindings/i2c/i2c-controller.yaml:12]: Why modifying this file?

This will affect **all** i2c bindings. Unless you have a good reason, don't do. And if you have one, make it its own commit with a self-explanatory commit message.

**Review by tbursztyka — COMMENTED** [review]
Overall, it seems fine but for one dts modification.

File: dts/bindings/i2c/i2c-controller.yaml (hunk 1 above)
Comment (finikorg) [reply dts/bindings/i2c/i2c-controller.yaml:12]: I think this is done because a new child device introduced with DMA settings. Not sure this is right way. How you combine i2-devices and this DMA child device together?

Comment (finikorg) [reply dts/bindings/i2c/i2c-controller.yaml:12]: Could you check that you can build this?
```
Found devicetree overlay: XXXX/zephyrproject/zephyr/samples/sensor/bme280/boards/intel_rpl_s_crb.overlay
devicetree error: 'reg' property in <Node /pcie0/i2c0/bme280@76 in 'XXXX/zephyrproject/zephyr/misc/empty_file.c'> has length 4, which is not evenly divisible by 12 (= 4*(<#address-cells> (= 2) + <#size-cells> (= 1))). Note that #*-cells properties come either from the parent node or from the controller (in the case of 'interrupts').
```

Comment (tbursztyka) [reply dts/bindings/i2c/i2c-controller.yaml:12]: indeed, that's not going to work, merging dma nodes and child.

Comment (tbursztyka) [reply dts/bindings/i2c/i2c-controller.yaml:12]: see #60374 for a possible solution

**Review by teburd — DISMISSED** [review]

File: dts/x86/intel/raptor_lake.dtsi [hunk 2]
```diff
@@ -57,103 +57,127 @@
...
-			#size-cells = <0>;
 			vendor-id = <0x8086>;
 			device-id = <0x7acc>;
 			interrupts = <PCIE_IRQ_DETECT IRQ_TYPE_LOWEST_LEVEL_LOW 3>;
 			interrupt-parent = <&intc>;
+			i2c_dma0: dma {
+				compatible = "intel,lpss";
+				#dma-cells = <1>;
+				status = "okay";
+			};
```
Comment (finikorg) [inline dts/x86/intel/raptor_lake.dtsi:68]: This would conflict with I2C device.

**Review by finikorg — CHANGES_REQUESTED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by finikorg — APPROVED** [review]
